sycl/source/platform.cpp

```diff
@@ -56,19 +56,47 @@ std::vector<platform> platform::get_platforms() {
 
 backend platform::get_backend() const noexcept { return impl->getBackend(); }
 
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+template <typename Param>
+typename detail::ABINeutralT<
+    typename detail::is_platform_info_desc<Param>::return_type>::type
+platform::get_info_impl() const {
+  auto Info = impl->template get_info<Param>();
+  if constexpr (std::is_same_v<decltype(Info), std::string>) {
+    return detail::string{Info};
+  } else if constexpr (std::is_same_v<decltype(Info),
+                                      std::vector<std::string>>) {
+    std::vector<detail::string> Res;
+    Res.reserve(Info.size());
+    for (std::string &Str : Info) {
+      Res.push_back(detail::string{Str});
+    }
+    return Res;
+  } else {
+    return Info;
+  }
```

Comment:
Please avoid code duplication with these to/from converters.

Reply:
Done. Thanks.